### PR TITLE
[Merged by Bors] - Add pull policy

### LINKED ISSUE #69: Add imagePullPolicy to SparkApplication CRD
As a user I want to be able to configure the pull-policy for `SparkApplication` job images, so that custom/job-specific images can be re-built and - if required - automatically reloaded into the Kubernetes cluster.

This issue will cover:

- add and handle an optional field imagePullPolicy to the SparkApplication CRD

### PR DESCRIPTION
**Pull request by adwk67**
# Description

Added two optional fields to the custom resource to govern pull policy and secrets to be used when accessing images.
Closes #69.

## Review Checklist

- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by razvan — COMMENTED** [review]
The image pull policy should also be added to the driver and executor pod templates.

**Comment by adwk67** [comment]
Isn't it the same image, though? The Job pod pulls the image (or not), and then this image is the same one used for the driver/executors (the pull policy only applies to the spark image, intentionally).

**Comment by razvan** [comment]
> Isn't it the same image, though? The Job pod pulls the image (or not), and then this image is the same one used for the driver/executors (the pull policy only applies to the spark image, intentionally).

Yes, but the Job starts one pod on one node, and in a production cluster there will be more nodes for running the actual spark cluster.

File: rust/operator-binary/src/spark_k8s_controller.rs [hunk 1]
```diff
@@ -208,31 +220,31 @@ fn pod_template(
...
     }
-    template.build().context(PodTemplateSnafu)
+
+    let pod = Pod {
+        metadata: ObjectMeta::default(),
+        spec: Some(PodSpec {
+            containers: vec![container.build()],
+            init_containers: inits,
+            volumes: Some(volumes),
+            image_pull_secrets: spark_application.spark_image_pull_secrets(),
```
Comment (razvan) [inline rust/operator-binary/src/spark_k8s_controller.rs:241]: Just a nitpick: this is now always set, unlike `image_pull_policy` which is set only if the user has configured it.

**Review by razvan — APPROVED** [review]
The tests still work and there is one small thing but it LGTM otherwise.

File: rust/operator-binary/src/spark_k8s_controller.rs (hunk 1 above)
Comment (adwk67) [reply rust/operator-binary/src/spark_k8s_controller.rs]: Implemented as suggested.

**Comment by adwk67** [comment]
bors merge

**Pull request closed without merging**